### PR TITLE
implement CinemaHalls model, dao, service and test it in main

### PR DESCRIPTION
**Pull request by IwalkInSilence**
implement MovieSession model, dao and service and do the same

### CODE REVIEW

File: src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllCinemaHallsQuery = session.createQuery(
+                    "from CinemaHall ", CinemaHall.class);
+            return getAllCinemaHallsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find all Movies in DB", e);
+        }
+    }
+}
+
```
Comment (Artificial-Friend) [inline src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java]: redundant empty line

File: src/main/java/project/cinema/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private CinemaHall cinemaHall;
+    private LocalDateTime showTime;
```
Comment (Artificial-Friend) [inline src/main/java/project/cinema/model/MovieSession.java:27]: Use @Column(name) here

**Review by Artificial-Friend — APPROVED** [review]
Good!

File: src/main/java/project/cinema/model/MovieSession.java (hunk 2 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/model/MovieSession.java:23]: but if I use it all broken up ...

File: src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllCinemaHallsQuery = session.createQuery(
+                    "from CinemaHall ", CinemaHall.class);
+            return getAllCinemaHallsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find all Movies in DB", e);
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java]: make this informative please

File: src/main/java/project/cinema/model/MovieSession.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private CinemaHall cinemaHall;
```
Comment (KseniiaMakarova) [inline src/main/java/project/cinema/model/MovieSession.java:25]: here let's use `@JoinColumn` to make the column be named `cinema_hall_id`

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 3 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/dao/impl/CinemaHallDaoImpl.java]: fix

File: src/main/java/project/cinema/model/MovieSession.java (hunk 4 above)
Comment (IwalkInSilence) [reply src/main/java/project/cinema/model/MovieSession.java:25]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**